plugins/uefi-capsule/fu-uefi-cod-device.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-uefi-cod-device.h"
+#include "fu-uefi-common.h"
+
+struct _FuUefiCodDevice
+{
+	FuUefiDevice		 parent_instance;
+};
+
+G_DEFINE_TYPE (FuUefiCodDevice, fu_uefi_cod_device, FU_TYPE_UEFI_DEVICE)
+
+static gboolean
+fu_uefi_device_get_results (FuDevice *device, GError **error)
+{
+	FuUefiDevice *device_uefi = FU_UEFI_DEVICE (device);
+	const gchar *tmp;
+	fwupd_guid_t guid = { 0x0 };
+	gsize bufsz = 0;
+	guint32 status = 0;
+	guint32 total_size = 0;
+	g_autofree gchar *err_msg = NULL;
+	g_autofree gchar *guidstr = NULL;
+	g_autofree gchar *version_str = NULL;
+	g_autofree guint8 *buf = NULL;
+
+	/* lets assume we just did one update */
+	if (!fu_efivar_get_data (FU_EFIVAR_GUID_EFI_GLOBAL, "Capsule0000",
```

Comment:
seems like a broad assumption, would it make more sense to store the variable name that was used into the history database or something?

Reply:
You can do easier than that. 
Start a loop and append the 'i' to Capsule.  So you'll read
Capsule0001
Capsule0002 etc, 
Once you fail to read a variable, you know your parsing needs to stop

Reply:
I've added something like this; can you verify it does the right thing? We might need to add another error in the `g_error_matches()` block.

Reply:
@hughsie, sure I'll try to run the whole thing tomorrow, but it seems to be doing the right thing

Reply:
You can also read the CapsuleLast variable (which will contain the name of the last applied capsule e.g Capsule0001) and figure out the size of 'idx'. The later is more 'efish' but both will work.

I did manage to run the whole thing and the ro/rw scanning of efivarfs is working correctly. The capsules are installed and executed and Capsule0000 is creted.  Here's the get-history output 

`Loading…                 [-                                      ]07:01:34:0315 FuEngine             failed to add device /sys/devices/platform/emmc2bus/fe340000.emmc2/mmc_host/mmc0/mmc0:0007/block/mmcb )
Loading…                 [***************************************]
WARNING: This package has not been validated, it may not work properly.
Unknown Product
│
└─Unknown Firmware:
  │   Device ID:          a45df35ac0e948ee180fe216a5f703f32dda163f
  │   Previous version:   0
  │   Update State:       Success
  │   Last modified:      2021-07-23 06:41
  │   GUID:               e2bb9c06-70e9-4b14-97a3-5a7913176e3f
  │   Device Flags:       • Internal device
  │                       • Updatable
  │                       • System requires external power source
  │                       • Needs a reboot after installation
  │                       • Device is usable for the duration of the update
  │ 
  └─  New version:      1.0.0
        License:          Unknown
`

Reply:
CapsuleLast contains the string Capsule0001? Could you please upload a tarball with the exact Capsule0001 and CapsuleLast contents and I'll add them to git and add some tests.

Reply:
I generated some for the tests.